File: src/Java/CreationalPatterns/AbstractFactory/EraStyles/src/ConcreteFactories/ClothingFactory.java
package ConcreteFactories;

import Abstraction.EraObject;
import Abstraction.EraObjectStylesFactory;
import Products.MedievalClothing;
import Products.RenaissanceClothing;
import Products.VictorianClothing;

public class ClothingFactory extends EraObjectStylesFactory {
    @Override
    public EraObject createMedievalStyleObject() {
        return new MedievalClothing();
    }

    @Override
    public EraObject createRenaissanceStyleObject() {
        return new RenaissanceClothing();
    }

    @Override
    public EraObject createVictorianEraStyleObject() {
        return new VictorianClothing();
    }
}

File: src/C#/CreationalPatterns/AbstractFactory/EraStyles/ConcreteFactories/ClothingFactory.cs
using EraStyles.Abstraction;
using EraStyles.Products;
using System;
using System.Collections.Generic;
using System.Text;

namespace EraStyles.ConcreteFactories
{
    public class ClothingFactory : EraObjectStylesFactory
    {
        public override EraObject CreateMedievalStyleObject()
        {
            return new MedievalClothing();
        }

        public override EraObject CreateRenaissanceStyleObject()
        {
            return new RenaissanceClothing();
        }

        public override EraObject CreateVictorianEraStyleObject()
        {
            return new VictorianClothing();
        }
    }
}
